src/main/java/org/study/DAO/impl/AdministratorDAOImpl.java

```diff
@@ -0,0 +1,226 @@
+     * and the substitute this parameter in user table column role_id
+     *
+     * @param administrator
+     */
+    @Override
+    public void create(Administrator administrator) {
+        PreparedStatement preparedStatement1 = null;
+        ResultSet resultSet1 = null;
+        PreparedStatement preparedStatement2 = null;
+        Connection con = null;
+        try {
+            con = INSTANCE;
+            preparedStatement1 = con.prepareStatement(ROLE_SQL_GETID);
+            preparedStatement1.setString(1, administrator.getUserRole().name().toLowerCase());
+            resultSet1 = preparedStatement1.executeQuery();
+            resultSet1.next();
+            int roleId = resultSet1.getInt("id");
+            preparedStatement2 = con.prepareStatement(ADMIN_SQL_CREATE);
+            setInsertValues(preparedStatement2, administrator, roleId);
+            preparedStatement2.executeUpdate();
+            LOG.info("Committing data here...");
+            TransactionManager.beginCommit();
+            LOG.info("The transaction was successfully");
+            LOG.info("New administrator was added");
+        } catch (SQLException e) {
+            LOG.error("Transaction failed, rollback the transaction", e);
+            TransactionManager.endTransaction();
+        } finally {
+            try {
+                if (resultSet1 != null) {
+                    resultSet1.close();
+                }
+            } catch (SQLException e) {
+                LOG.error("Result Set closing is failed", e);
+            }
+            try {
+                if (preparedStatement1 != null) {
+                    preparedStatement1.close();
+                }
+            } catch (SQLException e) {
+                LOG.error("Prepared Statement closing is failed", e);
+            }
+            try {
+                if (preparedStatement2 != null) {
+                    preparedStatement2.close();
+                }
+            } catch (SQLException e) {
+                LOG.error("Prepared Statement closing is failed", e);
+            }
+        }
+    }
+
+    /**
+     * this method update admin in database, this method firstly get user role id for role type
+     * (in case of roles extension this method would work correctly)
+     * and the substitute this parameter in user table column role_id
+     *
+     * @param administrator
+     */
+
+    @Override
+    public void update(Administrator administrator) {
+        PreparedStatement preparedStatement1 = null;
+        ResultSet resultSet1 = null;
+        PreparedStatement preparedStatement2 = null;
+        Connection con = null;
+        try {
+            con = INSTANCE;
+            preparedStatement1 = con.prepareStatement(ROLE_SQL_GETID);
+            preparedStatement1.setString(1, administrator.getUserRole().name().toLowerCase());
+            resultSet1 = preparedStatement1.executeQuery();
```

Comment:
Advice to review Dao structure. hard to read and understand

Reply:
Will be fix in next MR